File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.cpp
class Solution {
public:
    vector<int> findArray(vector<int>& pref) {
        vector <int> a (pref.size());

    a.front() = pref.front();
    for (int i = 1;i < pref.size(); i++){
        a[i] ^= (pref[i] ^ pref[i - 1]);
    }
        return a;
    }
};

File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.cs
public class Solution {
    public int[] FindArray(int[] pref) {
        int [] a = new int [pref.Length];
        
        a[0] = pref[0];
        for (int i = 1;i < pref?.Length; i++){
            a[i] ^= (pref[i] ^ pref[i - 1]);
        }
        return a;
    }
}